common/resource.go

```diff
@@ -366,7 +366,7 @@ func genericDatabricksData[T, P, C any](
 	hasOther bool) Resource {
 	var dummy T
 	var other P
-	otherFields := StructToSchema(other, NoCustomize)
+	otherFields := StructToSchema(other, nil)
```

Comment:
Why do we need this change?

Reply:
@alexott 
This change would allow passing  the `other`  as an instance of `ResourceProvider`, allowing for customization of selected fields in schema.

`ResourceProvider` carries its own customization function. The `StructToSchema` refactored in #3207 panics if a customization function is passed to it together with a schema of type `ResourceProvider`. The `StructToSchema` is pasted below for convenience:

```
func StructToSchema(v any, customize func(map[string]*schema.Schema) map[string]*schema.Schema) map[string]*schema.Schema {
	// If the input 'v' is an instance of ResourceProvider, call resourceProviderStructToSchema instead.
	if rp, ok := v.(ResourceProvider); ok {
		if customize != nil {
			panic("customize should be nil if the input implements the ResourceProvider interface; use CustomizeSchema of ResourceProvider instead")
		}
		return resourceProviderStructToSchema(rp)
	}
	rv := reflect.ValueOf(v)
	scm := typeToSchema(rv, map[string]string{})
	if customize != nil {
		scm = customize(scm)
	}
	return scm
}
```
The `NoCustomize` function returns the schema it gets without any changes, which is also what happens if `nil` is passed to `StructToSchema` as a customize function.